src/coreclr/jit/promotiondecomposition.cpp

```diff
@@ -523,16 +523,17 @@ class DecompositionPlan
         target_ssize_t addrBaseOffs       = 0;
         FieldSeq*      addrBaseOffsFldSeq = nullptr;
         GenTreeFlags   indirFlags         = GTF_EMPTY;
-
+        GenTreeFlags   flagsToPropagate   = GTF_IND_COPYABLE_FLAGS;
         if (m_store->OperIs(GT_STORE_BLK))
         {
+            flagsToPropagate |= GTF_IND_TGT_NOT_HEAP | GTF_IND_TGT_HEAP;
             addr       = m_store->AsIndir()->Addr();
-            indirFlags = m_store->gtFlags & GTF_IND_COPYABLE_FLAGS;
+            indirFlags = m_store->gtFlags & flagsToPropagate;
```

Comment:
Is there any benefit to adding something like
```cpp
if (m_store->AsBlk()->GetLayout()->IsStackOnly())
  indirFlags |= GTF_IND_TGT_NOT_HEAP;
```
here (and in morphblock.cpp)? Or will `m_store` already have this flag set for this case?

Reply:
@jakobbotsch I don't see diffs, but this PR already produces too many missing contexts 🙂 
Presumably, this is not enough, because some GT_STORE_BLK have this flag without its layout being IsStackOnly. E.g. when we store a non-byreflike struct to byreflike-struct's field

Reply:
Yeah, it makes sense to me to propagate the flags, but it also makes sense to me to set the flags anew here, since the decomposition is "losing" the information at this point. I will leave it up to you.

Reply:
Added. Looks like MihuBot found a few diffs with it